pyccl/core.py

```diff
@@ -577,35 +569,67 @@ def _build_parameters(
 
         # Fill an array with the non-relativistic neutrino masses
         if N_nu_mass > 0:
-            mnu_final_list = [0]*N_nu_mass
+            nu_mass = [0]*N_nu_mass
             relativistic = [0]*3
             for i in range(0, N_nu_mass):
                 for j in range(0, 3):
                     if (mnu_list[j] > 0.00017 and relativistic[j] == 0):
                         relativistic[j] = 1
-                        mnu_final_list[i] = mnu_list[j]
+                        nu_mass[i] = mnu_list[j]
                         break
         else:
-            mnu_final_list = [0.]
-
-        # Create new instance of ccl_parameters object
-        # Create an internal status variable; needed to check massive neutrino
-        # integral.
-        status = 0
-        if nz_mg == -1:
-            # Create ccl_parameters without modified growth
-            self._params, status = lib.parameters_create_nu(
-                Omega_c, Omega_b, Omega_k, Neff, w0, wa, h, A_s, sigma8, n_s,
-                T_CMB, Omega_g, T_ncdm, bcm_log10Mc, bcm_etab, bcm_ks,
-                mu_0, sigma_0, c1_mg, c2_mg, lambda_mg, mnu_final_list, status)
+            nu_mass = [0.]
+
+        c = const
+        # Fixed radiation parameters: (Omega_g h^2) is known from T_CMB.
+        rho_g = 4 * c.STBOLTZ / c.CLIGHT**3 * T_CMB**4
+        rho_crit = c.RHO_CRITICAL * c.SOLAR_MASS / c.MPC_TO_METER**3 * h**2
+
+        # Get the N_nu_rel from Neff and N_nu_mass.
+        N_nu_rel = Neff - N_nu_mass * T_ncdm**4 / (4/11)**(4/3)
+
+        # Temperature of the relativistic neutrinos in K.
+        T_nu = T_CMB * (4/11)**(1/3)
+        rho_nu_rel = N_nu_rel * (7/8) * 4 * c.STBOLTZ / c.CLIGHT**3 * T_nu**4
+        Omega_nu_rel = rho_nu_rel / rho_crit
+
+        # For non-relativistic neutrinos, calculate the phase-space integral.
+        self._fill_params(m_nu=nu_mass)
+        from .neutrinos import Omega_nu_h2
+        kw = {"m_nu": nu_mass, "T_CMB": T_CMB, "T_ncdm": T_ncdm}
+        Omega_nu_mass = Omega_nu_h2(1., **kw) / h**2 if N_nu_mass > 0 else 0
+        # Omega_nu_mass = self.OmNuh2(1.,) / h**2 if N_nu_mass > 0 else 0
```

Comment:
Note to self: have to check why replacing with the final line doesn't work.

Reply:
In which way doesn't it work?

Reply:
It returns a different value for OmNuh2, I have to check why.

Reply:
Fixed it.

Reply:
Embarrassing! Thanks!

Reply:
In the end it looks like `Omega_nu_mass` is ever so slightly different.